### PR TITLE
[Protocol3] Test signature-based whitelist implementation

### PR DESCRIPTION
**Pull request by yueawang**

### CODE REVIEW

**Comment by dong77** [comment]
也可以看看 https://github.com/Loopring/protocols/pull/594

File: packages/loopring_v3/test/testExchangeWhiteList.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
+import BN = require("bn.js");
+import { expectThrow } from "./expectThrow";
+import { Constants, Bitstream } from "loopringV3.js";
+import { ExchangeTestUtil } from "./testExchangeUtil";
```
[the remark this answers is not in the document]
Comment (yueawang) [reply packages/loopring_v3/test/testExchangeWhiteList.ts]: OK，单元测试是应该简单一点。
那集成测试还需要保留么？？或者有另外的test suite入口，后面可以加到那边去。

[the remark this answers is not in the document]
Comment (dong77) [reply packages/loopring_v3/test/testExchangeWhiteList.ts]: 不需要保留。因为exchange使用stakingPool的方法都是比较简单的逻辑。不需要那么严格测试了。

File: packages/loopring_v3/contracts/impl/SignatureBasedAddressWhitelist.sol [hunk 2]
```diff
@@ -26,6 +26,7 @@ import "../iface/IAddressWhitelist.sol";
 contract SignatureBasedAddressWhitelist is Claimable, IAddressWhitelist
 {
     uint public constant PERMISSION_TIMEOUT = 24 hours;
+    bytes SIG_PREFIX = "\x19Ethereum Signed Message:\n32";
```
Comment (dong77) [inline packages/loopring_v3/contracts/impl/SignatureBasedAddressWhitelist.sol]: 这个是不是要加constant？

File: packages/loopring_v3/contracts/impl/SignatureBasedAddressWhitelist.sol [hunk 3]
```diff
@@ -56,15 +58,18 @@ contract SignatureBasedAddressWhitelist is Claimable, IAddressWhitelist
...
         if (v < 27) {
             v += 27;
         }
-
         if (v != 27 && v != 28) {
             return false;
         }
-
-        bytes32 hash = keccak256(abi.encode("LOOPRING_DEX_ACCOUNT_CREATION", addr, t));
+        bytes32 msgBase = keccak256(abi.encodePacked("LOOPRING_DEX_ACCOUNT_CREATION", addr, t));
```
Comment (dong77) [inline packages/loopring_v3/contracts/impl/SignatureBasedAddressWhitelist.sol]: "LOOPRING_DEX_ACCOUNT_CREATION"
another constant?

Comment (yueawang) [reply packages/loopring_v3/contracts/impl/SignatureBasedAddressWhitelist.sol]: Yes, I think it's better not to use a constant member variable, but use this kind of constant string.

Comment (yueawang) [reply packages/loopring_v3/contracts/impl/SignatureBasedAddressWhitelist.sol]: So this time I removed the const string member but use
`bytes32 hash = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", msgBase));`
instead.

**Pull request merged**